packages/mui-material/src/Slider/useSlider.ts

```diff
@@ -714,7 +714,7 @@ export function useSlider(parameters: UseSliderParameters): UseSliderReturnValue
       type: 'range',
       min: parameters.min,
       max: parameters.max,
-      step: parameters.step === null && parameters.marks ? 'any' : parameters.step ?? undefined,
+      step: parameters.step === null && parameters.marks ? 'any' : (parameters.step ?? undefined),
```

Comment:
Unrelated change? If that's the case, let's undo it.

Reply:
#43276  seems to solve this issue.